### ISSUE TITLE
Automatically add padding to `FullGridCellList` bounding box?

### ISSUE THREAD

**Issue by efaulhaber** [trixi-framework/PointNeighbors.jl#80]
When a particle is in a boundary cell of the grid, it is not surrounded by cells, so we get an out-of-bounds exception when looking for neighbor cells. Currently, one has to manually add a padding of `search_radius` to the bounding box.
I think it would be better if the padding was automatically included (only when not using periodicity?) and maybe could be deactivated with `padding = false`.

**Comment by sloede** [comment]
Good idea! Maybe the padding could be automatically based on the maximum search radius (I don't know if this can be specified) in case of variable search radius setups?

**Comment by efaulhaber** [comment]
If we want to do variable search radius with p4est, we don't need a padding.

**Comment by efaulhaber** [comment]
Done in #90.
